sklearn/linear_model/tests/test_passive_aggressive.py

```diff
@@ -125,6 +126,77 @@ def test_classifier_undefined_methods():
         assert_raises(AttributeError, lambda x: getattr(clf, x), meth)
 
 
+def test_class_weights():
+    # Test class weights.
```

Comment:
Most of it, with some mods here and there. Some others might have been adapted from d-tree's tests if I recall correctly.